### PR TITLE
feat: create complete profile availability component

### PR DESCRIPTION
**Pull request by Lilly-nn**
## Describe your changes

- I created 'complete profile availability' component. If the user doesn't select a day of a week and time, the button is disabled:
image
image
image
image
When the user clicks on the selected day, time which he/she chose is displayed and it can be changed:
image

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://homecareaid.atlassian.net/browse/AC-99)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [ ] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [ ] Date and time formats are on the constants
- [ ] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

**Review by ProggerOleg — APPROVED** [review]

File: src/components/complete-profile-second/CompleteProfileSecond.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  BtnWrapper,
+  Container,
+  NextBtn,
+  SelectContainer,
+  WeekSlot,
+  WeekSlotContainer,
+  Wrapper,
+} from './styles';
+
+export default function CompleteProfileSecond({ onNext }: { onNext: () => void }): JSX.Element {
```
Comment (nadiia-tsytsylina) [inline src/components/complete-profile-second/CompleteProfileSecond.tsx:20]: Why it's complete profile second, if it's fourth?)
Снимок экрана 2023-11-24 194107

File: src/components/complete-profile-second/styles.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
+import { Typography, styled } from '@mui/material';
+import { PRIMARY, SECONDARY } from 'src/theme/colors';
+import { TYPOGRAPHY } from 'src/theme/fonts';
+import typography from 'src/theme/typography';
+import { FilledButton } from '../reusable';
```
Comment (nadiia-tsytsylina) [inline src/components/complete-profile-second/styles.ts:5]: I think, better to use absolute import in this case

File: src/components/complete-profile-second/CompleteProfileSecond.tsx (hunk 1 above)
Comment (Lilly-nn) [reply src/components/complete-profile-second/CompleteProfileSecond.tsx:20]: Oh, that's right

File: src/locales/langs/en.ts [hunk 3]
```diff
@@ -156,6 +157,11 @@ const en = {
       invalid_code: "Entered code doesn't match the code which was sent to your email.",
     },
   },
+  completeProfileSecond: {
```
Comment (nadiia-tsytsylina) [inline src/locales/langs/en.ts:160]: and here too

**Review by nadiia-tsytsylina — APPROVED** [review]

File: src/components/complete-profile-fourth/CompleteProfileFourth.tsx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+export default function CompleteProfileFourth({ onNext }: { onNext: () => void }): JSX.Element {
+  const { translate } = useLocales();
+
+  const availableDays = useTypedSelector((state) => state.availableDays);
+  const dispatch = useAppDispatch();
+
+  const [daySelected, setDaySelected] = useState<daySelectedType | null>(null);
+  const [chosenDays, setChosenDays] = useState<string[]>([]);
+
```
Comment (lyudad) [inline src/components/complete-profile-fourth/CompleteProfileFourth.tsx:28]: тут не нужна пустая строка, стейт к стейту

File: src/components/complete-profile-fourth/CompleteProfileFourth.tsx [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+        chooseAvailableTime({
+          time: {
+            from: availableFrom,
+            to: availableTo,
+          },
+          weekDay: daySelected,
+        })
+      );
+    }
+  }, [daySelected, availableFrom, availableTo, dispatch]);
```
Comment (lyudad) [inline src/components/complete-profile-fourth/CompleteProfileFourth.tsx:70]: юз ефект должен идти до хендлеров

**Review by lyudad — APPROVED** [review]

**Pull request merged**